easybuild/framework/easyblock.py

```diff
@@ -4751,8 +4751,8 @@ def make_checksum_lines(checksums, indent_level):
         if app.src:
             placeholder = '# PLACEHOLDER FOR SOURCES/PATCHES WITH CHECKSUMS'
 
-            # grab raw lines for source_urls, sources, patches
-            keys = ['patches', 'source_urls', 'sources']
+            # grab raw lines for the following params
+            keys = ['source_urls', 'sources', 'crates', 'patches']
```

Comment:
Hmm, this approach is a bit weird, as it breaks the boundary between framework and easyblocks, since `crates` is not a general easyconfig parameter...

Thoughts on this @Micket?

Reply:
Maybe we come up with an approach that would also work for `components` (in `Bundle` easyblock)?

Reply:
As discussed in the confcall I added a method `src_parameter_names` to the easyblock class.

In the `Cargo` easyblock we can override this to also return "crates".

For `components` I don't think this is useful. For now they look like:
```
components = [
    (name, version, {
        'source_urls': ['https://poppler.freedesktop.org/'],
        'sources': [SOURCE_TAR_XZ],
        'checksums': ['86b09e5a02de40081a3916ef8711c5128eaf4b1fc59d5f87d0ec66f04f595db4'],
        'configopts': "-DENABLE_BOOST=ON",
    }),
    ('poppler-data', '0.4.10', {
        'source_urls': ['https://poppler.freedesktop.org/'],
        'sources': [SOURCE_TAR_GZ],
        'checksums': ['6e2fcef66ec8c44625f94292ccf8af9f1d918b410d5aa69c274ce67387967b30'],
    }),
]
```

While the `Bundle` easyblock seemingly supports specifying `checksums` for everything at the top-level (outside the `components`) I don't think it actually does: It creates a full list of sources, patches and checksums during `__init__` but the actual checksum check (at least the contrib check) requires them in each component as above.

However with this design it would be possible to add the checksum list after the components but then we'd need to include logic to remove them from each component

The general approach is different here anyway: `crates` are just another way to specify `sources` while `components` are similar to `exts_list` which yield actual easyblock instances.